lib/ansible/modules/cloud/amazon/cloudfront_origin_access_identity.py

```diff
@@ -0,0 +1,285 @@
+
+- cloudfront_origin_access_identity:
+    state: present
+    caller_reference: this is an example reference
+    comment: this is an example comment
+
+# update an existing origin access identity
+# using caller_reference as an identifier
+
+- cloudfront_origin_access_identity:
+     state: present
+     origin_access_identity_id: E17DRN9XUOAHZX
+     caller_reference: this is an example reference
+     comment: this is a new comment
+
+# delete an existing origin access identity
+# using caller_reference as an identifier
+
+- cloudfront_origin_access_identity:
+     state: absent
+     caller_reference: this is an example reference
+     comment: this is a new comment
+
+'''
+
+RETURN = '''
+
+location:
+    description: describes a url specifying the origin access identity.
+    returned: always
+    type: str
+
+'''
+
+from ansible.module_utils.ec2 import get_aws_connection_info, ec2_argument_spec
+from ansible.module_utils.ec2 import boto3_conn, HAS_BOTO3
+from ansible.modules.cloud.amazon.cloudfront_facts import (
+    CloudFrontFactsServiceManager)
+import ansible.module_utils.cloudfront as helpers
+from ansible.module_utils.ec2 import camel_dict_to_snake_dict
+from ansible.module_utils.basic import AnsibleModule
+from botocore.signers import CloudFrontSigner
+import datetime
+from functools import partial
+import json
+import traceback
+
+try:
+    import botocore
+except ImportError:
+    pass
+
+
+class CloudFrontOriginAccessIdentityServiceManager(object):
+    """
+    Handles cloudfront origin access identity service calls to aws
+    """
+
+    def __init__(self, module):
+        self.module = module
+        self.create_client('cloudfront')
+
+    def create_client(self, resource):
+        try:
+            region, ec2_url, aws_connect_kwargs = get_aws_connection_info(
+                self.module, boto3=True)
+            self.client = boto3_conn(
+                self.module, conn_type='client', resource=resource,
+                region=region, endpoint=ec2_url, **aws_connect_kwargs)
+        except botocore.exceptions.NoRegionError:
+            self.module.fail_json(
+                msg=("region must be specified as a parameter in "
+                     "AWS_DEFAULT_REGION environment variable or in "
+                     "boto configuration file"))
+        except botocore.exceptions.ClientError as e:
+            self.module.fail_json(
+                msg="unable to establish connection - " + str(e),
+                exception=traceback.format_exc(),
+                **camel_dict_to_snake_dict(e.response))
+
+    def create_origin_access_identity(self, caller_reference, comment):
+        try:
+            return self.client.create_cloud_front_origin_access_identity(
+                CloudFrontOriginAccessIdentityConfig={
+                    'CallerReference': caller_reference,
+                    'Comment': comment
+                })
+        except botocore.exceptions.ClientError as e:
```

Comment:
BotoCoreError should be caught here as well and all other places catching ClientError https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/amazon/GUIDELINES.md#boto3-1

BotoCoreError does not have a .response, so using AnsibleAWSModule will cut down on exception handling logic as I suggested above.